src/vsphere_cpi/lib/cloud/vsphere/vm_creator.rb

```diff
@@ -114,6 +114,36 @@ def create(vm_config)
           raise "Unable to parse vmx options: 'vmx_options' is not a Hash"
         end
 
+        host = nil
+        # Before cloning we need to make sure the host is correctly picked up
+        # from cluster's host group if provided.
+        unless cluster.host_group.nil?
```

Comment:
this can be just if cluster.host_group

Reply:
I would prefer explicit nil? everywhere as it differentiates `false` from `nil` .  makes it clear that host_group is not a boolean. It does not sees so from the name but practice has to be consistent across variables.